lib/dao.js

```diff
@@ -1562,7 +1565,23 @@ DataAccessObject._coerce = function(where) {
 
       continue;
     }
+
     var DataType = props[p] && props[p].type;
+    if (p.match(/\./)) {
+      var model = p.split('.')[0];
+      var prop = p.split('.').slice(1);
+
+      if (props[model]) {
+        var clause = {};
+        clause[prop] = where[p];
+        where[p] = Array.isArray(props[model].type) ?
+          self._coerce(clause, props[model].type[0].definition.properties)[prop] :
+          self._coerce(clause, props[model].type.definition.properties)[prop];
```

Comment:
In that case, the test for querying on nested date that I added covers this and fails with the current implementation.